Eggong/Eggong/Presentation/Component/ActionButton.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  ActionButton.swift
+//  Eggong
+//
+//  Created by 조우현 on 10/20/24.
+//
+
+import SwiftUI
+
+struct ActionButton: View {
+    
+    enum State {
+        case enabled
+        case disabled
+        
+        var foreground: Color {
+            switch self {
+            case .enabled: return .black
+            case .disabled: return .white
+            }
+        }
+        
+        // TODO: 컬러시스템 확정 이후 컬러변경
+        var background: Color {
+            switch self {
+            case .enabled: return .yellow
+            case .disabled: return .gray
+            }
+        }
+    }
```

Comment:
S1
enum을 써서 깔끔하게 정리한 것도 좋은데, 코드가 좀 복잡해지는 느낌이 있네요 저한테는!
또, 런타임 중에 state를 바꿀 일도 있을 것 같아서 @State 형태의 프로퍼티로 써줘야 할 것 같아요!..아래와 같이 연산 프로퍼티를 활용하는건 어떨까요?

```suggestion
@Binding var enabled: Bool

var foreground: Color {
    return enabled ? .black : .white
}

var background: Color {
    return enabled ? .yellow : .gray
}
```